django/forms/forms.py

```diff
@@ -103,6 +109,16 @@ def __init__(self, data=None, files=None, auto_id='id_%s', prefix=None,
         self._bound_fields_cache = {}
         self.order_fields(self.field_order if field_order is None else field_order)
 
+        # Initialize form renderer. Use a global default if not specified
+        # either as an argument or as self.default_renderer.
+        if renderer is None:
```

Comment:
Looks like you don't ever handle the `else` case here. I would just change the next few lines of default-handling code to set `renderer` instead of `self.renderer`, and then follow up with an unconditional `self.renderer = renderer`